roles/lib_utils/src/generate.py

```diff
@@ -27,19 +27,47 @@ def parse_args():
     return parser.parse_args()
 
 
+def fragment_banner(fragment_path, side, data):
```

Comment:
I'm not sure why it's duplicated in both roles. I could probably even be, dare I say it, symlinked.